### PR TITLE
Improve network block validation process

### PR DESCRIPTION
**Pull request by rmoreliovlabs**
## Description

- Changed cidr bits (networkCidr) value from 24 to 16.
- Updated isAddressBlockAvailable to map addresses to InetAddressCidrBlock instead of InetAddressBlock.
- Removed decrepcated InetAddressBlock class.

## Motivation and Context
As of now, the config [maxConnections equals to 16](https://github.com/rsksmart/rskj/blob/2430282a73e86667d0e28dd2f6fe47ca956035de/rskj-core/src/main/resources/reference.conf#L129) and [network CIDR equals to 24](https://github.com/rsksmart/rskj/blob/2430282a73e86667d0e28dd2f6fe47ca956035de/rskj-core/src/main/resources/reference.conf#LL131C9-L131C25) by default. Long time ago maxConnections was change from 256 to 16 in scope of [this](https://github.com/rsksmart/rskj/pull/674/files#diff-242a8c6b5d9003504710db696af1f94412a52cdbd3369df55cff4be2ea8697eaL46-L51) PR. And no one knows why it was done at that time.

Due to this value is quite restrictive (especially when majority of our managed nodes are concentrated in AWS account), the idea is to relax these restrictions.

Goal #1: one option to relax it is to change CIDR block from 24 to 16, which would mean that a node with IP address X1.X2.X3.X4 would allow to connect max of 16 peers with address X1.X2.*.* 

Goal #2: to prevent confusions in future fix notation of networkCidr in a way that /8 corresponds to a block X1.*.*.* and /24 corresponds to X1.X2.X3.* respectively.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Requires Activation Code (Hard Fork)

* **Other information**:

### CODE REVIEW

**Comment by rmoreliovlabs** [comment]
pipeline:run

**Comment by rmoreliovlabs** [comment]
pipeline:run

**Comment by rmoreliovlabs** [comment]
pipeline:run

**Comment by rmoreliovlabs** [comment]
pipeline:run

**Comment by rmoreliovlabs** [comment]
pipeline:run

**Comment by rmoreliovlabs** [comment]
pipeline:run

**Comment by rmoreliovlabs** [comment]
pipeline:run

**Comment by rmoreliovlabs** [comment]
pipeline:run

File: rskj-core/src/test/java/org/ethereum/net/server/ChannelManagerImplTest.java [hunk 1]
```diff
@@ -70,10 +73,43 @@ void getNumberOfPeersToSendStatusTo() {
...
+        Channel otherPeer = new Channel(null, null, nodeManager, null, null, null, remoteId);
+        otherPeer.setInetSocketAddress(new InetSocketAddress("127.0.0.1",5554));
+
+        channelManager.add(peer);
+        channelManager.tryProcessNewPeers();
+
+        Logger logger = mock(Logger.class);
+        TestUtils.setFinalStatic(channelManager, "logger", logger);
+
+        channelManager.isAddressBlockAvailable(otherPeer.getInetSocketAddress().getAddress());
```
Comment (illuque-iov) [inline rskj-core/src/test/java/org/ethereum/net/server/ChannelManagerImplTest.java:108]: Maybe you should check that in this scenario this method returns `true`.

**Review by illuque-iov — APPROVED** [review]

File: rskj-core/src/test/java/org/ethereum/net/server/ChannelManagerImplTest.java (hunk 1 above)
Comment (rmoreliovlabs) [reply rskj-core/src/test/java/org/ethereum/net/server/ChannelManagerImplTest.java]: Done

**Comment by rmoreliovlabs** [comment]
pipeline:run

File: rskj-core/src/main/java/co/rsk/scoring/InetAddressUtils.java [hunk 2]
```diff
@@ -89,11 +86,15 @@ public static InetAddressCidrBlock parse(String text) throws InvalidInetAddressE
...
-        catch (NumberFormatException ex) {
+        } catch (NumberFormatException ex) {
             throw new InvalidInetAddressBlockException("Invalid mask", ex);
         }
 
+        return parse(address, nbits);
+    }
+
+
+    public static InetAddressCidrBlock parse(InetAddress address, int nbits) throws InvalidInetAddressException {
```
Comment (Vovchyk) [inline rskj-core/src/main/java/co/rsk/scoring/InetAddressUtils.java:97]:
```suggestion
    public static InetAddressCidrBlock createCidrBlock(InetAddress address, int nbits) throws InvalidInetAddressException {

```

**Review by Vovchyk — APPROVED** [review]

File: rskj-core/src/main/java/co/rsk/scoring/InetAddressUtils.java (hunk 2 above)
Comment (rmoreliovlabs) [reply rskj-core/src/main/java/co/rsk/scoring/InetAddressUtils.java]: Done

**Comment by rmoreliovlabs** [comment]
pipeline:run

**Review by Vovchyk — APPROVED** [review]

**Pull request merged**